src/App.js

```diff
@@ -0,0 +1,74 @@
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import NavBar from './components/NavBar';
+import testImg from './assets/images/test-img.png';
+import Photos from './pages/Photos';
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import NoResult from './components/NoResult';
+import Loader from './components/Loader';
+
+const API_URL = 'https://jsonplaceholder.typicode.com/albums/1/photos';
+
+const App = () => {
+  const [photos, setPhotos] = useState([]);
+  const [filteredValue, setFilteredValue] = useState('');
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const checkErrorsFromAPI = (response) => {
+    if(response.status !== 200) {
+      setIsError(true);
+      throw new Error(response.statusText);
+    }
+  };
+
+  useEffect(() => {
+    const getPhoto = () => {
+      axios.get(API_URL)
+      .then(response => {
+        setPhotos(response.data)
+      })
+      .catch(() => checkErrorsFromAPI())
+      .finally(() => setIsLoading(false));
+    } 
+    setIsLoading(true);
+    getPhoto();
+  }, [filteredValue]);
+
+  const filterData = (newFilteredValue) => {
+    setFilteredValue(newFilteredValue);
+  };
+
+  const getFilteredPhotos = (photos, filteredValue) => photos.filter(photoEl => photoEl.title.includes(filteredValue));
```

Comment:
Filtering needs to be done with API so we can filter all photos not only the ones displayed on current page.